docs/architecture/adr-007.md

```diff
@@ -10,88 +13,101 @@ The current way to define those events is not flexible: events are hardcoded, pa
 
 This can lead to potential issue in the long run when commands needs to be updated to add more parameters (ibc-go/ gravity bridge) or even if we need to implement more "actions". Since contracts are immutable, contract developers won't be able to have access to these functionalities after deploying their CRC20 contracts if we perform an upgrade.
 
-In this ADR, we define generic events format that can be use by smart contract developer to call any "action" upon successful contract execution. 
+In this ADR, we define generic event argument that can be extended to accept new parameters in the future. 
 
 ## Decision
 
-We define 2 specifics events that smart contract can use to trigger specific action after contract execution.
+### __CronosSendToChainV2 Event
 
-```
-# Trigger an action after contract execution which only involve the sender account
-__CosmosCall(string functionName, address sender,  byte[]args)
+#### Changes
 
-# Trigger an action after contract execution that require moving asset from the smart contract "account"
-__CosmosCallAfterPayment(string functionName, address payer, uint amount, byte[]args)
+- Introduce event `__CronosSendToChainV2`, successor of event `__CronosSendToChain`;
```

Comment:
reposting my comment,

Why adding a V2? we can keep the same name

Reply:
This is a design decision only.

Use the same event:
- Two format of the same event exist;
- Chain logic has to perform the argument existence and type check at runtime, doable but maybe more error prone

Upgrade event
- Event format is fixed, i.e. argument type remains unchanged;
- Separated chain logic and maybe less error prone

So I decide to favour the readability and clear chain logic.

Reply:
since `SendToIBC` to reintroduce in CRC21, I think we could keep the same name but having different parameters

CRC20 will be marked as deprecated and ask people to uses CRC21

We will still support both events, in the hook level, we could simply check the arguments

Reply:
Ok I will update this section.